Library/Homebrew/livecheck/livecheck.rb

```diff
@@ -66,7 +66,10 @@ def livecheck_formulae(formulae_to_check, args)
       @livecheck_strategy_names.freeze
 
       has_a_newer_upstream_version = false
-      formulae_checked = formulae_to_check.sort.map.with_index do |formula, i|
+      formulae_checked = formulae_and_casks_to_check.sort_by(&:name).map.with_index do |formula_or_cask, i|
```

Comment:
`sort_by(&:name)` doesn't work in the same way for casks as it does with formulae. A number of casks contain one or more `name` values and the `#name` value is an array when there's more than one. We encounter a `comparison of Array with String failed` error in that case.

Additionally, some of the `name`s in casks aren't in English and those end up being sorted at the end. This would seem unusual to users, since the `token` is printed in the output instead and would appear to be out of alphabetical order.

The general idea is that we need to be sorting by the `token` for casks and the `name` for formulae. It would probably make more sense to just do this in `Livecheck#livecheck`, something like:

```ruby
formulae_and_casks_to_check = if args.tap
  ...
end.sort do |a, b|
  a_name = a.respond_to?("token") ? a.token : a.name
  b_name = b.respond_to?("token") ? b.token : b.name
  a_name <=> b_name
end
```

[I'm not using `#formula_or_cask_name`, `#formula_name`, or `#cask_name` above because we may not want to sort by the full name.]

We can then modify this line to:

```suggestion
      formulae_checked = formulae_and_casks_to_check.map.with_index do |formula_or_cask, i|
```